thrust/system/cuda/detail/transform_scan.h

```diff
@@ -50,8 +50,8 @@ transform_inclusive_scan(execution_policy<Derived> &policy,
                          TransformOp                transform_op,
                          ScanOp                     scan_op)
 {
-  // Use the input iterator's value type per https://wg21.link/P0571
```

Comment:
Please restore this comment. It clarifies why we choose certain behavior for consistency in edge cases. Before there was a bit of a mismash of input value types, binary functor results, output iterators, initial value types, etc between the different algorithms and implementations.

Reply:
Done.